### ISSUE TITLE
Carbon throw exception InvalidArgumentException with message Trailing data on Illuminate\Database\Eloquent\Model::asDateTime

### ISSUE THREAD

**Issue by firodj** [laravel/framework#10121]
I am using Laravel 5.1, HHVM 3.8, Carbon 1.20, PostgreSQL 9.4.

I got InvalidArgumentException exception with message "Trailing data", when Query result setting attribute for field with timestamp type. The value like "2015-09-01 18:46:01.145664" makes Carbon throws exception. 

Because `Illuminate\Database\Eloquent\Model::asDateTime` only check `value` for only:
- Carbon instance
- DateTime instance
- is_numeric
- 'Y-m-d' format
- 'Y-m-d H:i:s' format

To solve the problem it should check for trailing US (microseconds), and do one of:
1. Change format into 'Y-m-d H:i:s.u' or 'Y-m-d H:i:s+', OR
2. Just strip the trailing which adequate the 'Y-m-d H:i:s' format,

The field was generated by Laravel 4.2 with type `timestamp` (with precision), which by some raw sql command it may possible to insert the timestamp value prefixed with precision. On new generated field using Laravel 5.1 this problem should not occurs because the field will have type `timestamp(0)` (withou precision).

**Comment by GrahamCampbell** [comment]
This is an issue with 4.2.

**Comment by hpawe01** [comment]
I also ran into this issue when I switched to Laravel 5.x.
I **fixed** this issue by performing the following **psql**-Query for every table that has timestamps:
```
ALTER TABLE [table] ALTER created_at TYPE timestamp(0) without time zone, ALTER updated_at TYPE timestamp(0) without time zone, ALTER deleted_at TYPE timestamp(0) without time zone;
```
This changed the column type from `timestamp without time zone` to `timestamp(0) without time zone` (as proposed by @firodj). Now it is working.

The issue (and solution) should be part of the ["Upgrading To 5.0 From 4.2"-Guide from laravel](http://laravel.com/docs/5.1/upgrade#upgrade-5.0).
